test/OpenSpaceToolkit/Physics/Environment.test.cpp

```diff
@@ -753,3 +757,144 @@ TEST (OpenSpaceToolkit_Physics_Environment, Test_3)
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST (OpenSpaceToolkit_Physics_Environment, EarthCreationTest)
+{
+
+    using ostk::core::types::Shared ;
+    using ostk::core::types::Integer ;
+    using ostk::core::ctnr::Array ;
+    using ostk::core::fs::Directory ;
+
+    using ostk::physics::time::Scale ;
+    using ostk::physics::time::Instant ;
+    using ostk::physics::time::DateTime ;
+    using ostk::physics::Environment ;
+    using ostk::physics::env::Object ;
+    using ostk::physics::env::obj::celest::Earth ;
+
+    {
+
+        const Instant instant = Instant::DateTime(DateTime(2018, 1, 1, 0, 0, 0), Scale::UTC) ;
+
+        const Array<Shared<Object>> objects =
+        {
+            std::make_shared<Earth>(Earth::EGM84())
+        } ;
+
+        EXPECT_NO_THROW(Environment environment(instant, objects) ;) ;
+
+    }
+
+    // [TBI] Add a test for the relevant parameters of each earth model at different levels of truncations
+
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST (OpenSpaceToolkit_Physics_Environment, SphericalEarthTest)
+{
+
+    using ostk::core::types::Shared ;
+    using ostk::core::types::Integer ;
+    using ostk::core::types::Real ;
+    using ostk::core::ctnr::Array ;
+    using ostk::core::fs::Directory ;
+
+    using ostk::math::obj::Vector3d ;
+
+    using ostk::physics::units::Length ;
+    using ostk::physics::units::Derived ;
+    using ostk::physics::units::Time ;
+    using ostk::physics::time::Scale ;
+    using ostk::physics::time::Instant ;
+    using ostk::physics::time::DateTime ;
+    using ostk::physics::Environment ;
+    using ostk::physics::env::Object ;
+    using ostk::physics::env::obj::Celestial ;
+    using ostk::physics::env::obj::celest::Earth ;
+    using ostk::physics::coord::Frame ;
+    using ostk::physics::coord::Position ;
+    using ostk::physics::data::Vector ;
+
+
+    {
+        static const Derived::Unit      GravitationalParameterSIUnit                    =   Derived::Unit::GravitationalParameter(Length::Unit::Meter, Time::Unit::Second) ;
+        // const Derived                   Earth::Models::Spherical::GravitationalParameter=       { 398600441500000.0, GravitationalParameterSIUnit } ;
+
+        const Instant instant = Instant::J2000() ;
+
+        const Array<Shared<Object>> objects =
+        {
+            std::make_shared<Earth>(Earth::Spherical())
+        } ;
+
+        Environment environment(instant, objects) ;
+
+        Shared<const Celestial> earthCelestial = environment.accessCelestialObjectWithName("Earth") ;
+
+        const Vector gravField = earthCelestial->getGravitationalFieldAt(Position::Meters(Vector3d(6371000.0,0.0,0.0), Frame::ITRF())) ;
```

Comment:
These lines are testing functions on `Celestial` and `Earth` specifically and are not specific to `Environment`, correct?